bugalink-backend/trips/views.py

```diff
@@ -81,3 +88,46 @@ def reject(self, request, *args, **kwargs):
         trip_request.status = "REJECTED"
         trip_request.save()
         return Response(self.get_serializer(trip_request).data)
+
+
+class TripRecommendationViewSet(
+    mixins.RetrieveModelMixin,
+    mixins.DestroyModelMixin,
+    viewsets.GenericViewSet,
+):
+    queryset = Trip.objects.all()
+    serializer_class = TripSerializer
+
+    @action(detail=True, methods=["get"])
+    def get(self, request, user_id, *args, **kwargs):
+        try:
+            data = {"trips": []}
+            for passenger_routine in PassengerRoutine.objects.filter(
+                passenger__user_id=user_id
+            ):
+                data["trips"] = (
+                    data["trips"]
+                    + PassengerRoutineSerializer(passenger_routine)
+                    .get_recommendations(obj=passenger_routine)
+                    .data
+                )
+
+            return Response(data, status=status.HTTP_200_OK)
+        except Exception as e:
+            return Response({"msg": str(e)}, status=status.HTTP_400_BAD_REQUEST)
+
+
+class TripByIdViewSet(
+    mixins.RetrieveModelMixin,
+    mixins.DestroyModelMixin,
+    viewsets.GenericViewSet,
+):
+    queryset = Trip.objects.all()
+    serializer_class = TripSerializer
+
+    def get(self, request, trip_id, *args, **kwargs):
+        try:
+            trip = Trip.objects.get(id=trip_id)
+            return Response(TripSerializer(trip).data, status=status.HTTP_200_OK)
+        except Exception as e:
+            return Response({"message": str(e)}, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
Again, the error returning should be handled with django, not  by creating a custom response that defaults to a HTTP 400 bad_request